src/services/dashboards/widgets/_hooks/use-widget-state.ts

```diff
@@ -98,26 +98,26 @@ export function useWidgetState<Data = any>(
 ) {
     const state = reactive<WidgetState<Data>>({
         widgetConfig: computed<WidgetConfig>(() => getWidgetConfig(props.widgetConfigId)),
-        title: computed<string>(() => props.title ?? state.widgetConfig.title),
+        title: computed(() => props.title ?? state.widgetConfig.title),
         options: computed<WidgetOptions>(() => getRefinedOptions(
             state.widgetConfig.options,
             props.options,
             props.inheritOptions,
             props.dashboardVariables,
         )),
-        currency: computed(() => state.settings.currency?.value ?? CURRENCY.USD),
-        groupBy: computed<GroupBy|string>(() => state.options?.group_by),
-        granularity: computed<Granularity>(() => state.options?.granularity),
+        currency: computed(() => state.settings?.currency?.value ?? CURRENCY.USD),
+        groupBy: computed(() => state.options?.group_by),
+        granularity: computed(() => state.options?.granularity),
         chartType: computed<ChartType|undefined>(() => state.options?.chart_type),
-        size: computed<WidgetSize>(() => {
-            if (state.widgetConfig.sizes.includes(props.size)) return props.size;
+        size: computed(() => {
+            if (props.size && state.widgetConfig.sizes.includes(props.size)) return props.size;
```

Comment:
I wonder why the type was deleted

Reply:
Since the `WidgetState` interface was defined separately, it was removed from inside the `state`.